### ISSUE TITLE
Issue when using esm import

### ISSUE THREAD

**Issue by ngohungphuc** [iamkun/dayjs#593]
**Describe the bug**
So I have two issue when using  "dayjs": "1.8.14"

The first one if I'm using import like this

```
import dayjs from "dayjs";
import utc from "dayjs/plugin/utc";
import LocalizedFormat from "dayjs/plugin/localizedFormat";

dayjs.utc(lockedAt).locale(localize) .format("lll");
```
My code is working perfectly fine but angular Cli will throw error 
Module "node_modules/dayjs/index" has no default export. 

Then I change my code to
````
import dayjs from "dayjs/esm";
import utc from "dayjs/esm/plugin/utc";
import LocalizedFormat from "dayjs/esm/plugin/localizedFormat";
````

Then the angular cli not throw any error but the code is not working and I have this error 

> Cannot read property 'formats' of undefined

What have i done wrong.

**Comment by xxyuk** [comment]
A re-porduction link or repo will be better

**Comment by antony** [comment]
I get the same issue. I think it might be related to this: https://github.com/iamkun/dayjs/issues/598

**Comment by BingeCode** [comment]
Also still having issues with the plugin imports in angular project. 
I can use most dayjs functions just fine with `import * as dayjs from 'dayjs'`.

But when I need to use plugins, I import with `import * as weekday from 'dayjs/plugin/weekday`, it throws me this error in the browser - even though the compiler recognises the function weekday just fine:

`dayjs__WEBPACK_IMPORTED_MODULE_5__(...).month(...).startOf(...).weekday is not a function`

![image](https://user-images.githubusercontent.com/48513535/59253781-3dd17700-8c2f-11e9-9c02-de2d97df229d.png)

**Comment by muuvmuuv** [comment]
I got the same issue with localization. At the very beginning of my GatsbyJS site I do:

```js
const dayjs = require('dayjs')
require('dayjs/locale/de')
dayjs.locale('en')
```

Inside my react script I tried these two:

```js
import dayjs from 'dayjs'

const date = new window.Date()
console.log(date) // Tue Oct 01 2019 00:00:00 GMT+0200 (Mitteleuropäische Sommerzeit)
console.log(dayjs(date)) // {$L: "en", $d: Tue Oct 01 2019 00:00:00 GMT+0200 (Mitteleuropäische Sommerzeit), $y: 2019, $M: 9, $D: 1, …}
console.log(dayjs(date).format()) // 2019-10-01T00:00:00+02:00
console.log(dayjs(date).format('LL')) // LL
console.log(dayjs(date).locale('de')) // {$L: "de", $d: Tue Oct 01 2019 00:00:00 GMT+0200 (Mitteleuropäische Sommerzeit), $y: 2019, $M: 9, $D: 1, …}
console.log(dayjs(date).format('LL')) // LL
console.log(dayjs(date).locale('de').format('LL')) // LL
```

This just outputs `LL` as a string, no date formatting. When I use `dayjs/esm` I get an error that the var weekdays is not set.

**Comment by ngohungphuc** [comment]
Solve by using "dayjs": "^1.8.16"

Add this in tsconfig.json

`"allowSyntheticDefaultImports": true,`

And use like this

```
import dayjs from "dayjs";
import utc from "dayjs/plugin/utc";
import LocalizedFormat from "dayjs/plugin/localizedFormat";

dayjs.extend(LocalizedFormat);
dayjs.extend(utc);
```

**Comment by revelt** [comment]
for posterity, for non-TS setups, if calling the `extend` (like `dayjs.extend(advancedFormat)`) breaks your app, the `dayjs` has to be excluded from webpack `externals`

**Comment by vittoriozamboni** [comment]
Got the same issue when using `import * as dayjs from 'dayjs';`.  Using `import dayjs from 'dayjs';` solved the issue for me, while keeping `dayjs` in webpack's externals.

**Comment by keul** [comment]
For posterity: I have the same issue and none of the fix above worked.

```javascript
import dayjs from 'dayjs';
import durationPlugin from 'dayjs/plugin/duration';
dayjs.extend(durationPlugin);
```

In my environment I'm adding `dayjs` and registering a plugin inside a webpack chunk (so inside a submodule loaded using code splitting).

No way to make it work, I was forced to move the registration in the main chunk.

**Comment by hitautodestruct** [comment]
If anyone is having a similar issue with node I just had to add the extension to the utc import.
My project is vanilla js and it kept trying to import the typescript version of the utc plugin.

I'm running nodejs 14.x

TL;DR
```
import dayjs from "dayjs";
import utc from "dayjs/plugin/utc.js"; // <------ Notice the .js

dayjs.extend(utc)
```

**Comment by NoelBaron** [comment]
> If anyone is having a similar issue with node I just had to add the extension to the utc import.
> My project is vanilla js and it kept trying to import the typescript version of the utc plugin.
> 
> I'm running nodejs 14.x
> 
> TL;DR
> 
> ```js
> import dayjs from "dayjs";
> import utc from "dayjs/plugin/utc.js"; // <------ Notice the .js
> 
> dayjs.extend(utc)
> ```

This is the correct answer for node

**Comment by Mhorvilleur13** [comment]
How can I export dayjs.extend(isToday)? Sorry I am a beginner and I am trying to export isToday() to multiple files.

**Comment by yonario** [comment]
> How can I export dayjs.extend(isToday)? Sorry I am a beginner and I am trying to export isToday() to multiple files.

Try stack overflow. Your question is off topic.

**Comment by alexkuc** [comment]
For me, I was able to get this working with the following:

> import dayjs from 'dayjs'

With TypeScript version `4.6.4` and DayJS version `1.11.3`.

I use Vite with the following TypeScript config:

```js
...
    "esModuleInterop": false,
    "allowSyntheticDefaultImports": true,
...
```
